inc/agent/communication/abstractrequest.class.php

```diff
@@ -332,22 +332,40 @@ protected function addToResponse(array $entries) {
     * @return void
     */
    private function addNode(DOMElement $parent, $name, $content) {
-      if (is_array($content)) {
-          $node = $parent->appendChild(
-              $this->response->createElement(
-                  $name
-              )
-          );
-         foreach ($content as $sname => $scontent) {
-            $this->addNode($node, $sname, $scontent);
+      if (is_array($content) && !isset($content['content']) && !isset($content['attributes'])) {
+         if (!is_string($name)) {
```

Comment:
What is the type of `$name` in this case ? Maybe it should be documented in the PHPDoc.

Reply:
Also, maybe it could be mutualized:
```php
$node = is_string($name)
   ? $parent->appendChild($this->response->createElement($name))
   : $parent;
foreach ($content as $sname => $scontent) {
   $this->addNode($parent, $sname, $scontent);
}
```

Reply:
> What is the type of `$name` in this case ? Maybe it should be documented in the PHPDoc.

This happens when `$name` is an array index (see https://github.com/glpi-project/glpi/pull/9789/files#diff-c94f64b54ddda2d5546092c3a1666d5c7d6035b60afa72190b1f710b04a7755dR144)